File: Poly.app/poly.dev/api/src/main/java/ru/polydev/poly_dev/entity/Questions.java
package ru.polydev.poly_dev.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "questions")
@Getter
@Setter
@NoArgsConstructor
public class Questions {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, unique = true)
    private Long id;

    @Column(name = "text", nullable = false)
    private String text;

    @Column(name = "is_optional", nullable = false)
    private Boolean isOptional;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;
}

File: Poly.app/poly.dev/app/lib/repositories/entity/questions.dart
class Questions {
  Questions({
    required this.id,
    required this.text,
    required this.isOptional,
    required this.isActive,
  });

  final int id;
  final String text;
  final bool isOptional;
  final bool isActive;

  factory Questions.fromJson(Map<String, dynamic> json) {
    return Questions(
      id: json['id'],
      text: json['text'],
      isOptional: json['isOptional'],
      isActive: json['isActive'],
    );
  }
}
